### PR TITLE
restore: rewrite auto increment id after pitr

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/46520

Problem Summary:

After we seperate `auto_increment` and `tidb_rowid`. pitr didn't handle the `tidb_rowid` correctly. This PR fixed it. and add a new restriction about TableVersionInfo. When `model.CurrLatestTableInfoVersion` updated. backup and restore need to check carefully about compatibility.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
the same steps as description is issue.
and check the result.

backup cluster:
image

restore cluster:
image
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fix the issue that pitr will lost auto-increment id allocator update after table rewrite during restore
```

### CODE REVIEW

File: meta/meta.go [hunk 1]
```diff
@@ -316,6 +316,26 @@ func (*Meta) autoIncrementIDKey(tableID int64) []byte {
 	return []byte(fmt.Sprintf("%s:%d", mIncIDPrefix, tableID))
```
Comment (Leavrth) [inline meta/meta.go:316]: call the `AutoIncrementIDKey` instead.

**Review by Leavrth — APPROVED** [review]

**Review by tiancaiamao — APPROVED** [review]

**Pull request merged**